common/djangoapps/student/models.py

```diff
@@ -3116,6 +3122,135 @@ class AccountRecoveryConfiguration(ConfigurationModel):
+    This class is for course agnostic celebrations (not specific to a particular enrollment).
+    CourseEnrollmentCelebration is for celebrations that happen separately for each separate course.
+
+    The first_day_of_streak, last_day_of_streak and last_streak_celebration fields are used to
+    control celebration of the streak feature.
+    A streak is when a learner visits the learning MFE N number of times.
+    The business logic of streaks for a 3 day streak and 1 day break is the following:
+    1. Each streak should be celebrated exactly once, once the learner has completed the streak.
+    2. If a learner misses enough days to count as a break, the streak resets back to 0.
+    3. The streak is measured against the learner's configured timezone
+    4. We keep track of the total length of the streak, so there is a possibility in the future
+    to add multiple celebrations for longer streaks.
+    5. We keep track of the longest_ever_streak field for potential future use for badging purposes.
+
+    .. no_pii:
+    """
+    user = models.OneToOneField(User, models.CASCADE, related_name='celebration')
+    first_day_of_streak = models.DateField(default=None, null=True, blank=True)
+    last_day_of_streak = models.DateField(default=None, null=True, blank=True)
+    last_streak_celebration = models.DateField(default=None, null=True, blank=True)
+    longest_ever_streak = models.IntegerField(default=0)
+
+    def __str__(self):
+        return (
+            "[CourseEnrollmentCelebration] user: {}; first_day_of_streak {};"
+            " last_day_of_streak {}; last_streak_celebration {}; longest_ever_streak {};"
+        ).format(self.user.username, self.first_day_of_streak,
+                 self.last_day_of_streak, self.last_streak_celebration, self.longest_ever_streak)
+
+    @classmethod
+    def _get_now(cls):
+        """ Retrieve the value for the current datetime in the user's timezone """
+        # importing here to avoid a circular import
+        from lms.djangoapps.courseware.context_processor import user_timezone_locale_prefs
+        user_timezone_locale = user_timezone_locale_prefs(crum.get_current_request())
+        user_timezone = timezone(user_timezone_locale['user_timezone'] or str(UTC))
+        return user_timezone.localize(datetime.now())
+
+    @classmethod
+    def _calculate_streak_updates(cls, celebration, now):
+        """ Calculate the updates that should me applied to the streak fields of the provided celebration """
+        first_day_of_streak = celebration.first_day_of_streak
+        last_day_of_streak = celebration.last_day_of_streak
+        last_streak_celebration = celebration.last_streak_celebration
+        today = now.date()
+        yesterday = today - timedelta(days=1)
+        should_celebrate = False
+
+        first_ever_streak = None is last_streak_celebration is first_day_of_streak is last_day_of_streak
+        in_active_streak = last_day_of_streak and first_day_of_streak
```

Comment:
This name is fine... but a quick scan makes it read as `inactive_streak` which is the opposite of what it's defining. Is there a word that gets this across but is harder to misread? Like... `streak_is_active` or `in_current_streak` or something?